### PR TITLE
add Hong Kong holidays

### PR DESCRIPTION
**Pull request by paulld**

### CODE REVIEW

**Comment by ptrimble** [comment]
Hi @paulld! Thanks for the PR! Don't worry about the failures, I know about this issue and I'm gonna be just dropping jruby 1.7 support shortly. It's old and we shouldn't support it.

@ttwo32 Can you also give this a look so we have two sets of eyes on it?

**Comment by ptrimble** [comment]
For future reference, `hk` matches the iso3166-2 spec: https://en.wikipedia.org/wiki/ISO_3166-2:HK

File: definitions/hk.yaml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    regions: [hk]
+    mday: 25
+    observed: to_monday_if_weekend(date)
+  - name: Boxing Day
+    regions: [hk]
+    mday: 26
+    observed: to_weekday_if_boxing_weekend(date)
+
+methods:
+  cn_new_lunar_day:
```
Comment (ptrimble) [inline definitions/hk.yaml]: 😍

**Comment by paulld** [comment]
@ttwo32 @ptrimble thanks for your review and no problem for holding it. this is only 50% complete anyway because half of the HK public holidays are based on the Chinese calendar. I tried adding the first day of the lunar new year using a very non-scientific way, the tests pass locally but seems travis disagrees, I'll look into it

File: definitions/hk.yaml (hunk 1 above)
Comment (ttwo32) [inline definitions/hk.yaml]: Fantastic!!Thank you for adding chinese new lunar_day.:smile:

**Comment by ttwo32** [comment]
Hi @paulld! Thanks for the PR ! 
As @ptrimble said, we will stop supporting JRuby1.7 shortly. 
Don,t worry about the failures.:smile:

**Comment by ptrimble** [comment]
I'm closing this and moving this over to this PR: https://github.com/holidays/definitions/pull/5

I gave the proper attribution. I'll handle generating the final definitions and also releasing v5.0.0.

Thanks again @paulld! Looking forward to the rest of the lunar dates. Maybe you should talk to @jonathanpike, who is also looking into various lunar holiday situations but for Korea. 😄

**Comment by n-studio** [comment]
@paulld I spotted some errors in the lunar dates (2023 and 2030), comparing with the wikipedia page https://en.wikipedia.org/wiki/Chinese_New_Year#Dates_in_Chinese_lunisolar_calendar. I'm trying to fix them but I don't know how to find when is the lunar new year for every year. What method did you use to get them?

**Pull request closed without merging**